### PR TITLE
Add SetMetadata for FormatContext

### PR DESCRIPTION
**Pull request by Mr3h1v**
#66 
support set metadata

### CODE REVIEW

File: format_context.go [hunk 1]
```diff
@@ -106,6 +106,13 @@ func (fc *FormatContext) Metadata() *Dictionary {
 	return newDictionaryFromC(fc.c.metadata)
 }
 
+func (fc *FormatContext) Metadata(d *Dictionary) {
```
Comment (asticode) [inline format_context.go:109]: Shouldn't it be `SetMetadata` instead?

File: format_context.go [hunk 2]
```diff
@@ -106,6 +106,13 @@ func (fc *FormatContext) Metadata() *Dictionary {
...
+	if d == nil {
```
Comment (asticode) [inline format_context.go:110]: I'd rather not ignore if `d == nil` in case someone would want to reset the metadata to `nil`

**Review by asticode — CHANGES_REQUESTED** [review]
Thanks for the PR 👍 

Could you also add a simple test [here](https://github.com/asticode/go-astiav/blob/master/format_context_test.go#L51), something like:
```go
d := NewDictionnary()
d.Set("k", "v", ...)
fc3.SetMetadata(d)
require.Equal(t, "v", fc3.Metadata().Get("k")
require.Equal(t, d
```

**Review by asticode — APPROVED** [review]

**Comment by asticode** [comment]
Thanks again for the PR ❤️ 

Let me know whether you need me to make a tag 👍

**Comment by Mr3h1v** [comment]
> Thanks again for the PR ❤️
> 
> Let me know whether you need me to make a tag 👍

I do need a tag. please make one

**Comment by asticode** [comment]
I've created a `v0.18.0` tag 👍

**Pull request merged**